ee/tabby-webserver/src/service/email.rs

```diff
@@ -76,22 +107,38 @@ pub async fn new_email_service(db: DbConn) -> Result<impl EmailService> {
 impl EmailService for EmailServiceImpl {
     async fn get_email_setting(&self) -> Result<Option<EmailSetting>> {
         let creds = self.db.read_email_setting().await?;
-        Ok(creds.map(Into::into))
+        let Some(creds) = creds else {
+            return Ok(None);
+        };
+        let creds = creds.try_into();
+        let Ok(creds) = creds else {
+            warn!(
+                "Email settings are corrupt, please inspect or reset them with deleteEmailSetting"
```

Comment:
let's delete the record on behalf of user. (call self.db.delete_email_setting here)